libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt

```diff
@@ -97,16 +107,36 @@ class MapboxOnboardRouter : Router {
             )
         ).build()
 
-        OfflineRouteRetrievalTask(navigatorNative, logger, object : OnOfflineRouteFoundCallback {
-            override fun onRouteFound(routes: List<DirectionsRoute>) {
-                callback.onResponse(routes)
+        retrieveRoute(offlineRouter.buildUrl(), callback)
+    }
+
+    override fun cancel() {
+        jobControl.scope.cancel()
+    }
+
+    private fun retrieveRoute(url: String, callback: Router.Callback) {
+        jobControl.scope.launch {
+            val routerResult = withContext(ThreadController.IODispatcher) {
```

Comment:
What about `ioJobController.scope.coroutineContext` instead of making `IODispatcher` `public`?

Reply:
I think we should not do it. In that case we don't need a scope, we just need to switch a context of current `main` scope. And it's confusing if we use dispatcher from one scope inside another one.
Dispatchers have to be public. We need to access them without scope object.

Reply:
Is there any risk of having `IODispatcher` `public`? Wondering if clients access it and use it could affect overall behavior on our side 🤔

Reply:
the same way customer can call `ThreadController.getMainScopeAndRootJob().scope.dispatcher` and access it